src/Network/MQTT/Client.hs

```diff
@@ -538,12 +548,16 @@ sendAndWait c@MQTTClient{..} dt f = do
 -- | Subscribe to a list of topic filters with their respective 'QoS'es.
 -- The accepted 'QoS'es are returned in the same order as requested.
 subscribe :: MQTTClient -> [(Filter, SubOptions)] -> [Property] -> IO ([Either SubErr QoS], [Property])
-subscribe c@MQTTClient{..} ls props = do
+subscribe c@MQTTClient{_cb} ls props = do
+  when (isOrdered _cb) runCallbackThread
```

Comment:
This is missing a guard against starting multiple callback threads (subscribe might be called multiple times).